lib/beaker/dsl/roles.rb

```diff
@@ -115,7 +115,11 @@ def not_controller(host)
       #       puts "this host is ONLY an agent!"
       #     end
       def agent_only(host)
-        host['roles'].length == 1 && host['roles'].include?('agent')
+        if host[:pe_ver] && version_is_less(host[:pe_ver], '4.0')
+          false
```

Comment:
it is an agent, it's just not "agent_only", which is a special kind of agent.

that does seem confusing, however.  Can we think of a better name for it?  That would be a much bigger diff, but I'm open to it if something more specific strikes.  `aio_agent?`, perhaps?